homeassistant/components/mqtt/client.py

```diff
@@ -376,6 +376,9 @@ def __init__(
 
         self._simple_subscriptions: dict[str, list[Subscription]] = {}
         self._wildcard_subscriptions: list[Subscription] = []
+        # _retained_init remembers we had an initial payload and should filter
+        # incoming retained payloads that are stale
+        self._retained_init: dict[Subscription, set[str]] = {}
```

Comment:
I think the comment and name of the dictionary is a bit misleading.
The dictionary tracks subscribers which have received a retained message on a certain topic.

Maybe:
```suggestion
        # _retained_topics prevents a Subscription from receiving a
        # retained message more than once per topic. This prevents flooding
        # already active subscribers when new subscribers subscribe to a topic
        # which has subscribed messages.
        self._retained_topics: dict[Subscription, set[str]] = {}
```